### PR TITLE
New button to add targets

### LINKED ISSUE #9: Add button in control bs4Card to add a new target into _targets.R bs4Card
## Prework

* [x] I understand and agree to `targetsketch`' [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to `targetsketch`' [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Proposal

Please describe the new feature. If applicable, write a minimal example in R code or pseudo-code to show input, usage, and desired output. To help us read any code you include (optional) please try to follow the [tidyverse style guide](https://style.tidyverse.org/). The `style_text()` and `style_file()` functions from the [`styler`](https://github.com/r-lib/styler) package make it easier.

Thanks for the package! Button in control card could open modal window where user can fill in new target info to add after the last target in `_targets.R`. User could select what arguments besides the required name and command they need. Tooltip could be display on mouse hover over argument. This would be helpful to orient new {targets} users to the available arguments for `tar_target()`. This package is a great learning tool for targets, similar to what {esquisse} provides for {ggplot2}. I can work on this if that's okay.

### PR DESCRIPTION
**Pull request by parmsam**
# Prework

* [x] I understand and agree to this repository's [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to this repository's [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] I have already submitted an issue to the [issue tracker](http://github.com/wlandau/targetsketch/issues) to discuss my idea with the maintainer.

# Summary

Please explain the purpose and scope of your contribution.

- Adds button to add new target to _targets.R card
- Adds tooltips
- Fixes #9 

# Related GitHub issues and pull requests

* Ref: #9

# Checklist

* [x] This pull request is not a [draft](https://github.blog/2019-02-14-introducing-draft-pull-requests).

### CODE REVIEW

File: R/server.R [hunk 1]
```diff
@@ -40,21 +40,52 @@ server <- function(input, output, session) {
...
         clipText = input$script,
-        icon = icon("clipboard")
+        icon = shiny::icon("clipboard")
       )
     })
   })
 
-  observeEvent(input$clipbtn, {
+  shiny::observeEvent(input$clipbtn, {
     shinyalert::shinyalert(title = "_targets.R copied to clipboard",
```
Comment (wlandau) [inline R/server.R:55]: Following the [tidyverse style guide advice on long lines](https://style.tidyverse.org/syntax.html?q=long%20lines#long-lines), would you create a new line for each of the arguments and indent each by 2 spaces?

File: R/server.R [hunk 2]
```diff
@@ -40,21 +40,52 @@ server <- function(input, output, session) {
...
+  })
+
+  shiny::observeEvent(input$modal_ok, {
+    print(input$modal_tar_name)
+    print(input$modal_tar_command)
+    if (nchar(input$modal_tar_name) > 1 &
+        nchar(input$modal_tar_command) > 1) {
+      original_text <- input$script
+      new_target_text <- paste0(
+        " |>\nappend(tar_target(",
```
Comment (wlandau) [inline R/server.R:70]: Following the [tidyverse style guide](https://style.tidyverse.org/pipes.html#whitespace) about whitespace in pipes, would you indent each new line by 2 spaces? On my end, there is no indent.

**Review by wlandau — COMMENTED** [review]
The pipe is such a slick way to approach this because the target list always goes last. Brilliant.

A few general comments:

* It's a small thing, but the "Declare a new target" menu "OK" button does not respond to my mouse clicks when either the target name or the target command is less than 2 characters long.
* Don't worry about the `R CMD check` issues, I think they were from my own commits.

File: R/server.R (hunk 2 above)
Comment (parmsam) [reply R/server.R]: Oh, I missed this. Fixed.

**Comment by wlandau** [comment]
One last request: would you find exported alternatives to `tools:::.Rd_get_text()` and `utils:::.getHelpFile()` to fix https://github.com/wlandau/targetsketch/runs/8309627545?check_suite_focus=true#step:6:80?

**Comment by parmsam** [comment]
I wasn't able to find exported alternatives when I looked around. It might be better to manually define the tooltip text for the name and command arguments.

File: R/utils.R [hunk 3]
```diff
@@ -38,8 +38,10 @@ label_with_tooltip <- function(label, ...) {
 get_function_doc <- function(function_name, package_name) {
   helptext <- utils::help(topic = (function_name),
                    package = (package_name))
-  help_documentation <- tools:::.Rd_get_text(
-    utils:::.getHelpFile(
+  rd_get_text <- utils::getFromNamespace(".Rd_get_text", "tools")
```
Comment (wlandau) [inline R/utils.R:41]: How necessary are these unexported functions? `getFromNamespace()` is still `:::` in essence, and coding on the internals of another package would put `targetsketch` in a vulnerable position.

Comment (wlandau) [reply R/utils.R:41]: Thanks. Would you do the same for `.Rd_get_text()`?

File: R/server.R [hunk 4]
```diff
@@ -73,3 +105,30 @@ update_values_impl <- function(values) {
...
+  shiny::showModal(
+    shiny::modalDialog(
+      shiny::textInput("modal_tar_name",
+                       label = label_with_tooltip(
+                         "Enter target name",
+                         paste0(
+                           tar_command_desc(),
+                           collapse = "\n")
+                         )),
+      shiny::textAreaInput("modal_tar_command",
```
Comment (wlandau) [inline R/server.R:119]: There are a few more places with large indents like this one. Would you start the arguments on new lines and indent by 2 spaces?

Comment (parmsam) [reply R/server.R:119]: Didn't realize I was doing that. Thanks for pointing that out. I'll fix that.

File: R/utils.R (hunk 3 above)
Comment (parmsam) [reply R/utils.R]: Please let me know if you would prefer tar_command_desc() and tar_name_desc() or the new version of get_function_doc(). Thanks!

Comment (wlandau) [reply R/utils.R]: I would prefer `tar_name_desc()` and `tar_command_desc()`. Either that or a short description that points to https://docs.ropensci.org/targets/reference/tar_target.html#arguments.

Comment (parmsam) [reply R/utils.R]: Sounds good. I've removed the get_function_doc().

**Pull request merged**